api/v1alpha1/contour_types.go

```diff
@@ -280,6 +302,27 @@ const (
 	GCPLoadBalancerProvider   LoadBalancerProviderType = "GCP"
 )
 
+// NodePort is the schema to specify a network port for a NodePort Service.
+type NodePort struct {
+	// Name is an IANA_SVC_NAME within the NodePort Service.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+
+	// PortNumber is the network port number to expose for the NodePort Service.
+	// The number must:
+	//
+	// 1. Not be used by another NodePort Service.
+	// 2. Be within the cluster's nodeport service range, i.e. --service-node-port-range
+	//    flag (default: 30000-32767).
+	// 3. Be a valid network port number, i.e. greater than 0 and less than 65536.
+	//
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	PortNumber int32 `json:"portNumber"`
```

Comment:
Is it possible to make this optional, i.e. to allow the Kubernetes cluster to choose the port number if not explicitly specified?

Reply:
Yes, I think that it would make the single-Contour case easier if this was defaulted to the current standard values maybe? Unless there's something we're trying to achieve by making this explicit?

Reply:
@skriss @youngnick I was trying to keep the default as-is, but I'm fine having the default select the nodeports dynamically.